packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -189,13 +232,24 @@ function newPotentialClickAction(
   }
 
   return {
+    base,
     addFrustration,
+    stop,
+
+    getFrustrations: () => frustrations,
+
+    onStop: (newOnStopCallback: () => void) => {
+      onStopCallback = newOnStopCallback
+    },
+
+    clone: () => newPotentialClickAction(lifeCycle, history, collectFrustrations, base),
 
     validate: (endTime?: TimeStamp) => {
-      if (isStopped) {
+      stop(endTime)
+      if (state.status !== PotentialClickActionStatus.STOPPED) {
         return
       }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about adding a test for this return?